src/sentry/tasks/post_process.py

```diff
@@ -336,53 +363,78 @@ def post_process_group(
 
             return
 
+        # group states
+        # {
+        #     "id": gi.group.id,
+        #     "is_new": gi.is_new,
+        #     "is_regression": gi.is_regression,
+        #     "is_new_group_environment": gi.is_new_group_environment,
+        # }
+
+        # ppg_configs = {
+        #     IssueCategory.Error: PostProcessGroupConfig(
+        #         post_process_functions=[
+        #             run_issue_alerts,
+        #             run_snooze,
+        #             run_issue_assignment,
+        #             ...
+        #         ]
+        #     )
+        # }
+
+        # config = ppg_configs[group.issue_category]
+        # # Do common stuff
+        # for ppg_func in post_process_functions:
+        #     ppg_func(group)
+
+        group_states = kwargs.get("group_states")
+
+        # error issue
+        if group_states is None:
+            group_states = {
+                1: {
+                    "id": group_id,
+                    "is_new": is_new,
+                    "is_regression": is_regression,
+                    "is_new_group_environment": is_new_group_environment,
+                }
+            }
+        else:
+            # TODO handle perf issues with multiple groups, not sure how to do update_event_group, handle_owner_assignment, etc
+            pass
+
+        # TODO does this work for multiple group ids?
         update_event_group(event)
```

Comment:
will need some help with this for multiple group ids

Reply:
I have a feeling that from this point forward we'll be iterating over a loop of all the `GroupEvent` entries from the event. That will be a separate pr though.

Reply:
So I'm trying to think this through a bit. I think that this should actually become a bulk function. So we'd pass the `event` and `group_states`. Then inside this we do:

```
# Ideally we'd be able to bulk fetch groups with redirects but for now we can do this serially
event.groups = [get_group_with_redirect(group_state["id"]) for group_state in group_states]
for group in event.groups:
    with sentry_sdk.start_span(op="tasks.post_process_group.fetch_buffered_group_stats"):
        # Ideally this could also happen in bulk
        fetch_buffered_group_stats(group)

    group.project = event.project
    group.project.set_cached_field_value("organization", event.project.organization)
````